### PR TITLE
Update Troubleshooting

### PR DESCRIPTION
**Pull request by justinegeffen**
Updated Troubleshooting structure and also included SAML error messages for this ticket: https://mattermost.atlassian.net/browse/MM-19860.

### CODE REVIEW

**Comment by justinegeffen** [comment]
Also related to: https://mattermost.atlassian.net/browse/MM-19770 and includes removing the troubleshooting guides from https://docs.mattermost.com/deployment/sso-saml-troubleshooting.html and https://docs.mattermost.com/deployment/sso-saml-adfs-msws2016.html, and consolidating them in 
https://docs.mattermost.com/install/troubleshooting.html.

File: source/deployment/sso-saml-ldapsync.rst [hunk 1]
```diff
@@ -8,8 +8,8 @@ In addition to configuring SAML sign-in, you can optionally configure synchroniz
 
 To configure SAML synchronization with AD/LDAP:
 
-1. Go to **System Console > SAML** in prior versions or **System Console** > **Authentication** > **SAML 2.0** in versions after 5.12 and set **Enable Synchronizing SAML Accounts With AD/LDAP** to `true`.
-2. Go to **System Console > AD/LDAP** in prior versions or **System Console** > **Authentication** > **AD/LDAP** in versions after 5.12 and set **Enable Synchronization with AD/LDAP** to `true`.
+1. Go to **System Console > Authentication > SAML 2.0** (or **System Console > SAML** in versions prior to 5.12) and set **Enable Synchronizing SAML Accounts With AD/LDAP** to `true`.
```
Comment (amyblais) [inline source/deployment/sso-saml-ldapsync.rst]:
```suggestion
1. Go to **System Console > Authentication > SAML 2.0** (or **System Console > SAML** in versions prior to 5.12) and set **Enable Synchronizing SAML Accounts With AD/LDAP** to ``true``.
```

File: source/deployment/sso-saml-ldapsync.rst [hunk 2]
```diff
@@ -8,8 +8,8 @@ In addition to configuring SAML sign-in, you can optionally configure synchroniz
...
+2. Go to  **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and set **Enable Synchronization with AD/LDAP** to `true`.
```
Comment (amyblais) [inline source/deployment/sso-saml-ldapsync.rst]:
```suggestion
2. Go to  **System Console > Authentication > AD/LDAP** (or **System Console > AD/LDAP** in versions prior to 5.12) and set **Enable Synchronization with AD/LDAP** to ``true``.
```

File: source/deployment/sso-saml-ldapsync.rst [hunk 3]
```diff
@@ -20,33 +20,32 @@ To configure SAML synchronization with AD/LDAP:
...
-3. Run AD/LDAP sync in **System Console > AD/LDAP > AD/LDAP Synchronize Now** in prior versions or **System Console** > **Authentication** > **AD/LDAP** > **AD/LDAP Synchronize Now** in versions after 5.12.
-4. Purge all caches again in Mattermost in **System Console > Configuration > Purge All Caches** in prior versions or **System Console** > **Web Server** > **Purge All Caches** in versions after 5.12, which re-activates the account in Mattermost.
+2. Purge all caches in Mattermost in **System Console > Web Server > Purge All Caches** (or **System Console > Configuration > Purge All Caches** in versions prior to 5.12).
+3. Run AD/LDAP sync in **System Console > Authentication > AD/LDAP > AD/LDAP Synchronize Now** (or **System Console > AD/LDAP > AD/LDAP Synchronize Now** in versions prior to 5.12).
+4. Purge all caches again in Mattermost in **System Console > Web Server > Purge All Caches** (or**System Console > Configuration > Purge All Caches** in versions prior to 5.12), which re-activates the account in Mattermost.
```
Comment (amyblais) [inline source/deployment/sso-saml-ldapsync.rst]:
```suggestion
4. Purge all caches again in Mattermost in **System Console > Web Server > Purge All Caches** (or **System Console > Configuration > Purge All Caches** in versions prior to 5.12), which re-activates the account in Mattermost.
```

File: source/install/troubleshooting.rst [hunk 4]
```diff
@@ -86,40 +84,193 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, ensure you have installed the `XML Security Library <https://www.aleksey.com/xmlsec/download.html>`__ on your Mattermost instance and that **it is available in your** PATH.
+
+Second, ensure you have completed each step of the SAML configuration.
+
+System Administrator locks themselves out of the system
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the System Administrator is locked out of the system during SAML configuration process, they can set an existing account to System Administrator using `a command line tool <http://docs.mattermost.com/deployment/on-boarding.html#creating-system-administrator-account-from-commandline>`__.
```
Comment (amyblais) [inline source/install/troubleshooting.rst]: Link is broken.

File: source/install/troubleshooting.rst [hunk 5]
```diff
@@ -86,40 +84,193 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
-Unable to Switch to SAML Authentication Successfully
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-First, ensure you have installed the `XML Security Library <https://www.aleksey.com/xmlsec/download.html>`__ on your Mattermost instance and that **it is available in your** PATH.
+``An error occurred while building Service Provider Metadata.``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This error indicates that the installation does not have an Enterprise license. The error message you receive will look similar to this:
+
+.. code-block:: sh
```
Comment (amyblais) [inline source/install/troubleshooting.rst]: Code block not rendering.

File: source/install/troubleshooting.rst [hunk 6]
```diff
@@ -86,40 +84,193 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+   ERRO[2019-12-23T09:06:27.654774-07:00] An error occurred while configuring SAML Service Provider  caller="app/enterprise.go:154" error="saml-public.crt: cannot read: failed to get config file saml-public.crt: failed to read file from /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-public.crt: open /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-public.crt: no such file or directory"
+
+Install the Service Provider Public Certificate and restart the process.
+
+.. note::
+
+   If making adjustments for these errors within System Console, not restart is required. However, if making configuration changes outside System Console, such as moving certificate files to the corrrect path.  A server restart is required.
```
Comment (amyblais) [inline source/install/troubleshooting.rst]:
```suggestion
   If making adjustments for these errors within System Console, no restart is required. However, if making configuration changes outside System Console, such as moving certificate files to the correct path, a server restart is required.
```

File: source/install/troubleshooting.rst [hunk 7]
```diff
@@ -86,40 +84,193 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+   ERRO[2019-12-20T17:20:24.3999581-07:00] Identity Provider Public Certificate File was not found. Please contact your System Administrator.  caller="mlog/log.go:175" err_details= err_where=SamlInterfaceImpl.BuildRequest http_code=500 ip_addr="::1" method=GET path=/login/sso/saml request_id=tm9ywzxcbj88dypkhjgg8hideo user_id=
+
+Install the Identity Provider Certificate and restart the process.
+
+**Issue: Missing Service Provider Private Key**
+
+.. code-block:: sh
+
+   ERRO[2019-12-23T08:51:28.423397-07:00] An error occurred while configuring SAML Service Provider  caller="app/enterprise.go:154" error="saml-public.crt: cannot read: failed to get config file saml-private.key: failed to read file from /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-private.key: open /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-private.key: no such file or directory"
```
Comment (sbishel) [inline source/install/troubleshooting.rst]: I realized you just copied the error messages, I gave you. But can we change the paths to ".../mattermost-server/config/saml-private.key". The beginning part of the path will depend on their install anyway.

File: source/install/troubleshooting.rst [hunk 8]
```diff
@@ -86,40 +84,193 @@ If email sign-in was turned off before the System Administrator switched sign-in
...
+
+Install the Service Provider Private Key and restart the process.
+
+**Issue: Missing Service Provider Public Certificate**
+
+.. code-block:: sh
+
+   ERRO[2019-12-23T09:06:27.654774-07:00] An error occurred while configuring SAML Service Provider  caller="app/enterprise.go:154" error="saml-public.crt: cannot read: failed to get config file saml-public.crt: failed to read file from /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-public.crt: open /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-public.crt: no such file or directory"
```
Comment (sbishel) [inline source/install/troubleshooting.rst]: Same, change paths to ".../mattermost-server/config/saml-public.crt".

File: source/install/troubleshooting.rst (hunk 7 above)
Comment (justinegeffen) [reply source/install/troubleshooting.rst]: Sure - I meant to ask you about that, so thanks for reminding me!

**Comment by justinegeffen** [comment]
@amyblais, the links are working in my local build (rendering and clickable) and I updated the URL. Could you let me know if they're working on your side now?

**Comment by amyblais** [comment]
The link is working now.

**Pull request merged**